tests/framework_starlette/test_bg_tasks.py

```diff
@@ -87,10 +87,10 @@ def _test():
         response = app.get("/" + route)
         assert response.status == 200
 
-    BUG_COMPLETELY_FIXED = (starlette_version >= (0, 21, 0)) or (
-        starlette_version >= (0, 20, 1) and sys.version_info[:2] > (3, 7)
+    BUG_COMPLETELY_FIXED = (0, 21, 0) <= starlette_version < (0, 23, 1) or (
+        (0, 20, 1) <= starlette_version < (0, 23, 1) and sys.version_info[:2] > (3, 7)
     )
-    BUG_PARTIALLY_FIXED = (0, 20, 1) <= starlette_version < (0, 21, 0) and sys.version_info[:2] <= (3, 7)
+    BUG_PARTIALLY_FIXED = (0, 20, 1) <= starlette_version < (0, 21, 0) or starlette_version >= (0, 23, 1)
```

Comment:
This doesn't seem to be backwards compatible with the previous logic. Is there a good reason for that or is there a mistake here?

Reply:
I realized the equal-to-or-greater-than Python 3.7 check is not necessary for our tests anymore since we removed testing for Python 3.6.